src/exec.c

```diff
@@ -634,15 +661,24 @@ int hyper_run_process(struct hyper_exec *exec)
 		goto close_tty;
 	}
 
+	if (pipe2(sync_pipe, O_CLOEXEC) < 0) {
+		perror("create sync_pipe failed");
+		goto close_tty;
+	}
+
 	pid = fork();
 	if (pid < 0) {
 		perror("fork prerequisite process failed");
 		goto close_tty;
 	} else if (pid == 0) {
-		hyper_do_exec_cmd(exec, pipe[1], &io);
+		hyper_do_exec_cmd(exec, pipe[1], sync_pipe[1], &io);
 	}
 	fprintf(stdout, "prerequisite process pid %d\n", pid);
 
+	// Close sync_pipe fd from the parent.
+	close(sync_pipe[1]);
```

Comment:
fd is initialized to -1, meaning the close will have no effect in case the fd has not been previously set.
About `close_if_set`, where does this function come from ? I mean which library ?